### PR TITLE
qip-0007.md: UTXO ledger specification

### PR DESCRIPTION
**Pull request by wizeguyy**

### CODE REVIEW

File: qip-0007.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+#### External Transactions
+The Quai network is a sharded system, where each 'zone chain' processes transactions related to its shard(s) of the ledger(s), as defined in [qip-0002](qip-00002.md). Furthermore, [qip-0005](qip-0005.md) describes how transactions may transmit value or data across shards, via an external transaction message (aka ETX). The 
+The mechanics of transmitting and referencing ETXs across shards/chains is described fully in [qip-0005](qip-0005.md). This document will just focus on the structure, processing, and emission of ETXs from a given shard of the Qu ledger.
+
+Simply, an ETX is equivalent to a Qu UTXO. When an ETX becomes referencable at the destination shard, each ETX will be directly added to the UTXO and made available for spending in the Qu ledger.
+
+Conversely, when emitting an ETX, the ETX may contain extra data according to the transaction format of the ledger it is destined for. When a Qu transaction creates an output to an address which resides outside the shard which processed the transaction, that output will be treated as an ETX instead of being added as a new output in the UTXO set.
```
[the remark this answers is not in the document]
Comment (kiltsonfire) [reply qip-0007.md:89]: By implication we might also consider limiting the number of inputs to a ETX to 1 but obviously it must not reference anything outside of scope.

**Review by kiltsonfire — CHANGES_REQUESTED** [review]

File: qip-0007.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      index: uint8,
+      signature: []byte,
+    },
+    ...
+  ],
+  outputs: [
+    {
+      version: uint8,
+      denomination: uint8,
+      address: []byte,
```
Comment (jdowning100) [inline qip-0007.md:71]: Is this the same as a public key script? i.e. the utxo is spendable by *this* address/public key

Comment (wizeguyy) [reply qip-0007.md:71]: This is the same address as used in the EVM ledger: a hash of the pubkey

File: qip-0007.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+| 15           | 100000 Qu    |
+
+### Transaction Structure
+Transactions will contain the following data, RLP encoded:
+```
+{
+  inputs: [
+    {
+      txhash: []byte,
+      index: uint8,
```
Comment (gameofpointers) [inline qip-0007.md:62]: what does index mean here? I dont see an explanation

File: qip-0007.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      signature: []byte,
+    },
+    ...
+  ],
+  outputs: [
+    {
+      version: uint8,
+      denomination: uint8,
```
Comment (gameofpointers) [inline qip-0007.md:70]: what about len ?

File: qip-0007.md (hunk 3 above)
Comment (gameofpointers) [reply qip-0007.md:62]: I think its a good idea to explain the input struct as well, similar to the outputs

File: qip-0007.md (hunk 4 above)
Comment (wizeguyy) [reply qip-0007.md:70]: This entire structure is RLP encoded, which removes the need to manually specify lengths

**Review by gameofpointers — APPROVED** [review]

**Review by Djadih — APPROVED** [review]

**Pull request merged**